ClrMD.Extensions/ClrMDExtensions.cs

```diff
@@ -59,18 +59,24 @@ public static IEnumerable<ClrDynamic> EnumerateDynamicObjects(this ClrHeap heap,
 
             HashSet<ClrType> set = new HashSet<ClrType>(castedTypes);
 
-            return from address in heap.EnumerateObjectAddresses()
+            return from address in heap.EnumerateObjects().Select(f => f.Address)
                    let type = heap.GetSafeObjectType(address)
                    where set.Contains(type)
                    select new ClrDynamic(address, type);
         }
 
         public static IEnumerable<ClrDynamic> EnumerateDynamicObjects(this ClrHeap heap, string typeName)
         {
+            var heapTypes = heap.EnumerateTypes();
+                //(from objects in heap.EnumerateObjects()
+            //group objects by objects.Type into tn 
+            // select (tn.Key)
+            //);
+
             if (!typeName.Contains("*"))
             {
                 var type = 
-                    (from t in heap.EnumerateTypes()
+                    (from t in heapTypes
```

Comment:
It's bad practice to iterate multiple times over an IEnumerable var, it's better to leave it like it was and call `heap.EnumerateTypes()` multiple times